geotags/views.py

```diff
@@ -47,16 +49,18 @@
             ORDER BY sounds_sound.created DESC
                      %(end)s"""
 
-def generate_json(sound_queryset):
-    # When using Django Sound queries:
-    #sounds_data = [[s.id, s.geotag.lat, s.geotag.lon] for s in sound_queryset]
+def generate_bytearray(sound_queryset):
+    # sounds as bytearray
+    packed_sounds = cStringIO.StringIO()
+    for sound in sound_queryset:
+        packed_sounds.write(struct.pack("i", sound.id))
+        packed_sounds.write(struct.pack("i", int(sound.lat*1000)))
```

Comment:
From a binary point of view: you are using 4-byte ints to pack the data. Those go from -2147483648 to 2147483647. You want to encode lat and lon (float), which as a maximum range go from -180 to 180. That means you can use a lot more precision. You can can use: 10000000. (i.e. 2147483647/180 = 11930464.7, round down to nearest "round number" gets you 10000000)